recipes/chat-main.tsx

```diff
@@ -0,0 +1,255 @@
+  const id = getId(userId);
+  username.set(name);
+  users.update({ [id]: name } as any);
+  console.log("[setUsername] userId:", id, "name:", name);
+}, { proxy: true });
+
+// User Session Recipe - Individual instance with local state
+export const UserSession = recipe<
+  UserSessionInput,
+  UserSessionResult
+>(
+  "User Chat Session",
+  ({ messages, users }) => {
+    const username = cell<string>("");
+    const userId = cell<string>("");
+
+    // UI reads: `messages.map(...)` renders the list reactively. No writes here;
+    // those happen via the `sendMessage` handler above.
+    return {
+      [NAME]: str`Chat Session` as any,
+      [UI]: (
+        <div>
+          <h2>Your Chat Session</h2>
+          <div>
+            <label>
+              Your User ID: {derive(userId, (k) => k || "(pending)")}
+            </label>
+          </div>
+          <div>
+            <h4>Set your display name</h4>
+            <common-send-message
+              name="Set"
+              placeholder="Choose a display name"
+              appearance="rounded"
+              onmessagesend={setUsername({ username, users, userId })}
+            />
+          </div>
+          <div
+            data-testid="current-username"
+            style={{ marginTop: "6px", fontSize: "12px", color: "#444" }}
+          >
+            Your current username is: {derive(
+              { u: username, id: userId },
+              ({ u, id }) => (u?.trim() || id || "(pending)"),
+            )}
+          </div>
+          <hr />
+          <div>
+            <h3>Chat Messages</h3>
+            <div style={{ display: "grid", gap: "8px" }}>
+              {messages.map((m) => (
+                <div style={{ display: "grid", gap: "2px" }}>
+                  <div
+                    data-testid="message-header"
+                    style={{ fontSize: "12px", color: "#666" }}
+                  >
+                    <b>
+                      {derive(
+                        { u: users, id: m.userId },
+                        ({ u, id }) => resolveDisplayName(u as any, id as any),
+                      )}
+                    </b>
+                    <span>· {derive(m.timestamp, formatTime)}</span>
+                  </div>
+                  <div style={{ fontSize: "14px" }}>{m.message}</div>
+                </div>
+              ))}
+            </div>
+          </div>
+          <div>
+            <common-send-message
+              name="Send"
+              placeholder="Type your message..."
+              appearance="rounded"
+              onmessagesend={sendMessage({
+                messages,
+                userId,
+                username,
+                users,
+              })}
+            />
+          </div>
+        </div>
+      ),
+      userId: derive(userId, (k) => k || "") as any,
+      username: username,
+    };
+  },
+);
+
+// Handler to create a new user session. We use `{ proxy: true }` so the handler
```

Comment:
Instead of that, you could set the type to `Cell<...>`, then you also get a handle on the reactive reference.